src/com/lftechnology/java/training/dipak/employeemanagement/controller/EmployeeController.java

```diff
@@ -0,0 +1,261 @@
+			fullName = sc.nextLine();
+			e.setFullName(fullName);
+
+			EmployeeService ed = ServiceFactory.getEmployeeService();
+
+			ed.terminateEmployee(e);
+
+		}
+		catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception type::{0}", ex);
+		}
+		return count;
+	}
+
+	/**
+	 * <p>This method takes the input for the filters and calls the viewEmployee
+	 * method. </p>
+	 * 
+	 * @param e
+	 * @param sc
+	 * @return
+	 */
+	public ResultSet viewEmployee(Employee e, Scanner sc) {
+
+		String fullName = null;
+		String department = null;
+		String address = null;
+		try {
+			LOGGER.info("Please enter the employee fullname::");
+			fullName = sc.nextLine();
+
+			LOGGER.info("Please enter the department::");
+			department = sc.nextLine();
+
+			LOGGER.info("Please enter the address::");
+			address = sc.nextLine();
+
+			e.setFullName(fullName);
+			e.setAddress(address);
+			e.setDepartment(department);
+
+			EmployeeService ed = ServiceFactory.getEmployeeService();
+
+			ed.viewEmployee(e);
+
+		}
+		catch (Exception ex) {
+			LOGGER.log(Level.INFO, "Exception type::{0}", ex);
+		}
+		return null;
+	}
+
+	// public int deleteEmployee(Employee e,Scanner sc) {
+	// String fullName=null;
+	// String department=null;
+	// String address=null;
+	// int count=0;
+	// try {
+	//
+	// LOGGER.info("Please enter the employee username::");
+	// fullName=sc.nextLine();
+	//
+	// LOGGER.info("Please enter the department::");
+	// department=sc.nextLine();
+	// LOGGER.info("Please enter the address::");
+	// address=sc.nextLine();
+	//
+	// e.setFullName(fullName);
+	// e.setAddress(address);
+	// e.setDepartment(department);
+	//
+	// EmployeeService ed=ServiceFactory.getEmployeeService();
+	//
+	// count=ed.deleteEmployee(e);
+	//
+	// }catch(Exception ex) {
+	// LOGGER.log(Level.INFO,"Exception type::{0}",ex);
+	// }
+	// return count;
+	// }
+
+	/**
+	 * <p>This method takes the input for the fields to be edited and calls
+	 * editEmployeeDetails in service layer.</p>
+	 * 
+	 * @param e
+	 * @param sc
+	 * @return e
+	 */
+	public Employee editEmployeeDetails(Employee e, Scanner sc) {
+
+		LOGGER.info("Welcome to the edit mode....");
+		String userName = null;
+		String password = null;
+		String fullName = null;
+		String department = null;
+		String address = null;
+		Console cnsl = null;
+		try {
+			cnsl = System.console();
+			LOGGER.info("Enter new username::");
+			userName = sc.nextLine();
+
+			LOGGER.info("Enter new password::");
+			char[] pwd = cnsl.readPassword();
+			password = String.valueOf(pwd);
+			// password=sc.nextLine();
+
+			LOGGER.info("Enter new fullname::");
+			fullName = sc.nextLine();
+
+			LOGGER.info("Enter new department::");
+			department = sc.nextLine();
+
+			LOGGER.info("Enter new address::");
+			address = sc.nextLine();
+
+			if (!(userName.equals(""))) {
+				e.setUserName(userName);
+			}
+			if (!(password.equals(""))) {
+				e.setPassword(password);
+			}
+			if (!(fullName.equals(""))) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1132)